File: src/main/java/com/wafflestudio/draft/model/Region.java
package com.wafflestudio.draft.model;

import com.vividsolutions.jts.geom.Polygon;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.util.List;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity
public class Region {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String depth1;

    private String depth2;

    private String depth3;

    private Polygon polygon;

    @Column(nullable = false)
    private String name;

    @OneToMany(mappedBy = "region", cascade = CascadeType.ALL)
    private List<User> users;
}

File: src/main/kotlin/com/wafflestudio/draft/model/Region.kt
package com.wafflestudio.draft.model

import com.wafflestudio.draft.dto.RegionDTO
import org.locationtech.jts.geom.MultiPolygon
import javax.persistence.*

@Entity
class Region(
        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        var id: Long? = null,
        var emdCode: Long? = null,
        var depth1: String? = null,
        var depth2: String? = null,
        var depth3: String? = null,

        @Column(unique = true)
        var name: String? = null,

        @Column(nullable = false, columnDefinition = "Geometry(MultiPolygon,5179)")
        var polygon: MultiPolygon? = null,

        @OneToMany(mappedBy = "region", cascade = [CascadeType.ALL], fetch = FetchType.LAZY)
        var users: MutableList<User> = mutableListOf(),

        @OneToMany(mappedBy = "region", cascade = [CascadeType.ALL], fetch = FetchType.LAZY)
        var courts: MutableList<Court> = mutableListOf()

) {
    constructor(id: Long?, emdCode: Long?, depth1: String?, depth2: String?, depth3: String?, name: String?)
            : this(id, emdCode, depth1, depth2, depth3, name, null)

    fun toResponse(): RegionDTO.Response {
        return RegionDTO.Response(id, depth1, depth2, depth3,name)
    }

    fun toResponseWithRooms(): RegionDTO.ResponseWithRooms {
        return RegionDTO.ResponseWithRooms(this)
    }
}